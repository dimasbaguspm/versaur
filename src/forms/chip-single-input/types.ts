import type {
  FieldsetHTMLAttributes,
  InputHTMLAttributes,
  ReactNode,
} from 'react'

/**
 * Props for the ChipSingleInput component
 */
export interface ChipSingleInputProps
  extends Omit<FieldsetHTMLAttributes<HTMLFieldSetElement>, 'onChange'> {
  /**
   * The size of the chip: 'sm', 'md' (default), or 'lg'
   */
  size?: 'sm' | 'md' | 'lg'
  /**
   * Label text to display above the chip group (rendered as legend)
   */
  label?: ReactNode
  /**
   * Whether the field is required (displays asterisk in legend)
   */
  required?: boolean
  /**
   * Helper text to display below the chip group
   */
  helperText?: ReactNode
  /**
   * Error message for invalid state
   */
  error?: ReactNode
  /**
   * The name attribute for the radio group - required for radio functionality
   */
  name: string
  /**
   * Current selected value (controlled component)
   * String representing selected option value
   */
  value?: string
  /**
   * Callback when value changes
   */
  onChange?: (value: string) => void
  /**
   * Whether the chip group is read-only
   */
  readOnly?: boolean
  /**
   * Maximum width for individual chips (enables text truncation)
   */
  maxWidth?: string
}

/**
 * Props for the ChipSingleInput.Option component
 */
export interface ChipSingleInputOptionProps
  extends Omit<InputHTMLAttributes<HTMLInputElement>, 'type' | 'name'> {
  /**
   * The option label content
   */
  children: ReactNode
  /**
   * The value for this chip option
   */
  value: string
}
