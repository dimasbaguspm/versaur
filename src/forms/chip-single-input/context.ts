import { createContext, useContext } from 'react'
import type { ChipSingleInputProps } from './types'

export interface ChipSingleInputContextValue {
  size?: ChipSingleInputProps['size']
  disabled?: boolean
  readOnly?: boolean
  error?: boolean
  name: string
  value?: string
  maxWidth?: string
  onChange?: (value: string) => void
}

export const ChipSingleInputContext =
  createContext<ChipSingleInputContextValue | null>(null)

export const useChipSingleInputContext = () => {
  const context = useContext(ChipSingleInputContext)
  if (!context) {
    throw new Error(
      'useChipSingleInputContext must be used within ChipSingleInputContext'
    )
  }
  return context
}
