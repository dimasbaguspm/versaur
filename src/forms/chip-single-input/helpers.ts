import { cva } from 'class-variance-authority'

export const chipSingleInputOptionVariants = cva(
  'inline-flex items-center bg-white transition-colors duration-200 cursor-pointer border border-border',
  {
    variants: {
      variant: {
        primary: [
          'hover:bg-primary-soft',
          'hover:border-primary-bold',
          'focus:border-primary-light',
          'data-[selected=true]:bg-primary',
          'data-[selected=true]:border-primary',
          'data-[selected=true]:text-white',
        ].join(' '),
        secondary: [
          'hover:bg-secondary-soft',
          'hover:border-secondary-bold',
          'focus:border-secondary-light',
          'data-[selected=true]:bg-secondary',
          'data-[selected=true]:border-secondary',
          'data-[selected=true]:text-white',
        ].join(' '),
        tertiary: [
          'hover:bg-tertiary-soft',
          'hover:border-tertiary-bold',
          'focus:border-tertiary-light',
          'data-[selected=true]:bg-tertiary',
          'data-[selected=true]:border-tertiary',
          'data-[selected=true]:text-white',
        ].join(' '),
        ghost: [
          'hover:bg-ghost-soft',
          'hover:border-ghost-bold',
          'focus:border-ghost-light',
          'data-[selected=true]:bg-ghost',
          'data-[selected=true]:border-ghost',
          'data-[selected=true]:text-white',
        ].join(' '),
        neutral: ['hover:bg-gray-soft', 'hover:border-gray-light'].join(' '),
      },
      shape: {
        circle: 'rounded-full',
        rounded: 'rounded-lg',
      },
      size: {
        sm: 'h-7 px-3 text-sm min-w-[2.25rem]',
        md: 'h-9 px-4 text-sm min-w-[2.5rem]',
        lg: 'h-10 px-8 text-lg min-w-[2.75rem]',
      },
    },
    defaultVariants: {
      variant: 'primary',
      shape: 'circle',
      size: 'sm',
    },
  }
)
