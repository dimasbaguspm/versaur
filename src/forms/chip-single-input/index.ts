export { ChipSingleInput } from './chip-single-input'
export type * from './types'
