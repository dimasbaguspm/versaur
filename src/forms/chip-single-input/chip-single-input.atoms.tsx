import React from 'react'
import { cn } from '@/utils/cn'
import { chipSingleInputOptionVariants } from './helpers'
import { useChipSingleInputContext } from './context'
import type { ChipSingleInputOptionProps } from './types'
import { Check } from 'lucide-react'

/**
 * ChipSingleInput.Option atom component
 *
 * Individual chip option for ChipSingleInput
 * Uses radio input pattern for single selection
 * Supports leading icon and animated tick/check
 */
export const ChipSingleInputOption = React.forwardRef<
  HTMLInputElement,
  ChipSingleInputOptionProps
>(
  (
    {
      children,
      className,
      disabled,
      id,
      value,
      check,
      defaultCheck = false,
      ...props
    },
    ref
  ) => {
    const context = useChipSingleInputContext()
    const generatedId = React.useId()
    const inputId = id || generatedId
    const isDisabled = disabled || context.disabled
    // Only use supported variants for styling
    const supportedVariants = [
      'primary',
      'secondary',
      'tertiary',
      'ghost',
      'neutral',
    ] as const
    type SupportedVariant = (typeof supportedVariants)[number]
    const variant: SupportedVariant = supportedVariants.includes(
      context.variant as SupportedVariant
    )
      ? (context.variant as SupportedVariant)
      : 'primary'
    const shape = context.shape || 'circle'
    const size = context.size || 'sm'

    // Check if this option is selected
    const isChecked = context.value === value

    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      if (!isDisabled) {
        // Only one value can be selected
        context.onChange?.(value)
      }
      // Always call the original onChange prop if provided
      props.onChange?.(event)
    }

    // Icon size map for each chip size
    const iconSize = size === 'lg' ? 20 : size === 'md' ? 18 : 16

    return (
      <>
        <input
          ref={ref}
          type='radio'
          id={inputId}
          name={context.name}
          value={value}
          checked={isChecked}
          disabled={isDisabled}
          onChange={handleChange}
          className='sr-only'
          {...props}
        />
        <label
          htmlFor={inputId}
          data-selected={isChecked}
          className={cn(
            chipSingleInputOptionVariants({
              variant,
              selected: isChecked,
              shape,
              size,
            }),
            isDisabled && 'opacity-50 cursor-not-allowed pointer-events-none',
            className
          )}
        >
          <span className='flex items-center'>
            {/* Custom check, default check, or nothing */}
            {(check || defaultCheck) && (
              <span
                className={cn(
                  'flex items-center',
                  'transition-transform transition-opacity duration-200',
                  isChecked
                    ? 'opacity-100 scale-100 mr-1'
                    : 'opacity-0 scale-75'
                )}
                aria-hidden='true'
              >
                {isChecked &&
                  (check ? (
                    check
                  ) : (
                    <Check
                      size={iconSize}
                      className={
                        variant === 'primary'
                          ? 'text-primary'
                          : variant === 'secondary'
                            ? 'text-secondary'
                            : variant === 'tertiary'
                              ? 'text-tertiary'
                              : variant === 'ghost'
                                ? 'text-ghost'
                                : variant === 'neutral'
                                  ? 'text-neutral'
                                  : 'text-primary'
                      }
                    />
                  ))}
              </span>
            )}
            {children}
          </span>
        </label>
      </>
    )
  }
)
