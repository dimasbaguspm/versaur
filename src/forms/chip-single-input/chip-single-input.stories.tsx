/**
 * ChipSingleInput Option shape, size, and custom check stories
 *
 * Demonstrates the new shape, size, and check icon API for ChipSingleInput.Option
 */

import { CheckCircle, Star } from 'lucide-react'

/**
 * ChipSingleInput group stories
 *
 * Demonstrates usage of ChipSingleInput for multiple selection using checkbox pattern
 * Includes accessibility and variant examples
 */
import { useState } from 'react'
import { ChipSingleInput } from './chip-single-input'
import type { ChipSingleInputProps } from './types'
import { Icon } from '@/primitive'

export default {
  title: 'Forms/ChipSingleInput',
  component: ChipSingleInput,
  tags: ['autodocs'],
}

export const Basic = () => {
  const [value, setValue] = useState<string>('cherry')
  return (
    <ChipSingleInput
      name='fruits'
      variant='primary'
      value={value}
      onChange={setValue}
      aria-label='Select fruits'
    >
      <ChipSingleInput.Option value='apple'>Apple</ChipSingleInput.Option>
      <ChipSingleInput.Option value='banana'>Banana</ChipSingleInput.Option>
      <ChipSingleInput.Option value='cherry'>Cherry</ChipSingleInput.Option>
    </ChipSingleInput>
  )
}

export const Disabled = () => {
  const [value, setValue] = useState<string>('banana')
  return (
    <ChipSingleInput
      name='fruits'
      value={value}
      onChange={setValue}
      disabled
      aria-label='Select fruits'
    >
      <ChipSingleInput.Option value='apple'>Apple</ChipSingleInput.Option>
      <ChipSingleInput.Option value='banana'>Banana</ChipSingleInput.Option>
      <ChipSingleInput.Option value='cherry'>Cherry</ChipSingleInput.Option>
    </ChipSingleInput>
  )
}

export const Variants = () => {
  const [value, setValue] = useState<string>()
  const variants = ['primary', 'secondary', 'tertiary', 'ghost']
  const labels = ['Coral', 'Sage', 'Mist', 'Slate']
  return (
    <div className='flex flex-wrap gap-3'>
      {variants.map((variant, i) => (
        <ChipSingleInput
          key={variant}
          name={variant}
          value={value}
          onChange={setValue}
          variant={variant as ChipSingleInputProps['variant']}
        >
          <ChipSingleInput.Option value={variant}>
            {labels[i]}
          </ChipSingleInput.Option>
        </ChipSingleInput>
      ))}
    </div>
  )
}

/**
 * Shape variations for ChipSingleInput.Option
 */
export const Shape = () => {
  const [value, setValue] = useState<string>()
  return (
    <div className='flex flex-col gap-4'>
      <div>
        <div className='mb-2 font-semibold'>Circle (default)</div>
        <ChipSingleInput
          name='circle-shape'
          value={value}
          onChange={setValue}
          shape='circle'
        >
          <ChipSingleInput.Option value='a'>Circle A</ChipSingleInput.Option>
          <ChipSingleInput.Option value='b'>Circle B</ChipSingleInput.Option>
        </ChipSingleInput>
      </div>
      <div>
        <div className='mb-2 font-semibold'>Rounded</div>
        <ChipSingleInput
          name='rounded-shape'
          value={value}
          onChange={setValue}
          shape='rounded'
        >
          <ChipSingleInput.Option value='c'>Rounded C</ChipSingleInput.Option>
          <ChipSingleInput.Option value='d'>Rounded D</ChipSingleInput.Option>
        </ChipSingleInput>
      </div>
    </div>
  )
}

/**
 * Size variations for ChipSingleInput.Option
 */
export const Sizes = () => {
  const [value, setValue] = useState<string>()
  return (
    <>
      <div className='mb-2 font-semibold'>Small (default)</div>
      <ChipSingleInput
        name='sizes-sm'
        value={value}
        onChange={setValue}
        size='sm'
      >
        <ChipSingleInput.Option value='sm'>Small</ChipSingleInput.Option>
        <ChipSingleInput.Option value='md'>Medium</ChipSingleInput.Option>
        <ChipSingleInput.Option value='lg'>Large</ChipSingleInput.Option>
      </ChipSingleInput>
      <div className='mb-2 font-semibold mt-4'>Medium</div>
      <ChipSingleInput
        name='sizes-md'
        value={value}
        onChange={setValue}
        size='md'
      >
        <ChipSingleInput.Option value='sm'>Small</ChipSingleInput.Option>
        <ChipSingleInput.Option value='md'>Medium</ChipSingleInput.Option>
        <ChipSingleInput.Option value='lg'>Large</ChipSingleInput.Option>
      </ChipSingleInput>
      <div className='mb-2 font-semibold mt-4'>Large</div>
      <ChipSingleInput
        name='sizes-lg'
        value={value}
        onChange={setValue}
        size='lg'
      >
        <ChipSingleInput.Option value='sm'>Small</ChipSingleInput.Option>
        <ChipSingleInput.Option value='md'>Medium</ChipSingleInput.Option>
        <ChipSingleInput.Option value='lg'>Large</ChipSingleInput.Option>
      </ChipSingleInput>
    </>
  )
}

/**
 * Custom and default check icon variations for ChipSingleInput.Option
 */
export const CheckIcon = () => {
  const [value, setValue] = useState<string>()
  return (
    <div className='flex flex-col gap-4'>
      <div>
        <div className='mb-2 font-semibold'>Custom Check Icon</div>
        <ChipSingleInput
          name='custom-check'
          value={value}
          onChange={setValue}
          variant={value === 'circle' ? 'primary' : 'secondary'}
        >
          <ChipSingleInput.Option value='star'>
            <Icon as={Star} color='inherit' size='sm' />
            Star
          </ChipSingleInput.Option>
          <ChipSingleInput.Option value='circle'>
            <Icon as={CheckCircle} color='inherit' size='sm' />
            Circle
          </ChipSingleInput.Option>
        </ChipSingleInput>
      </div>
      <div>
        <div className='mb-2 font-semibold'>No Check Icon</div>
        <ChipSingleInput name='no-check' value={value} onChange={setValue}>
          <ChipSingleInput.Option value='x'>No Check A</ChipSingleInput.Option>
          <ChipSingleInput.Option value='y'>No Check B</ChipSingleInput.Option>
        </ChipSingleInput>
      </div>
    </div>
  )
}
