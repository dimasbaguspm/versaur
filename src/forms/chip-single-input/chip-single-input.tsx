import React from 'react'
import { cn } from '@/utils/cn'
import {
  ChipSingleInputContext,
  type ChipSingleInputContextValue,
} from './context'
import { ChipSingleInputOption } from './chip-single-input.atoms'
import type { ChipSingleInputProps } from './types'

/**
 * ChipSingleInput component for Versaur UI
 *
 * Provides a group of checkbox chips for multiple selection
 * Controlled component pattern for React forms
 */
const ChipSingleInputRoot = React.forwardRef<
  HTMLDivElement,
  ChipSingleInputProps
>(
  (
    {
      variant = 'primary',
      shape = 'circle',
      size = 'sm',
      label,
      helperText,
      error,
      className,
      disabled,
      name,
      value,
      onChange,
      children,
      ...props
    },
    ref
  ) => {
    const hasError = Boolean(error)

    const contextValue = {
      variant,
      shape,
      size,
      disabled,
      error: hasError,
      name,
      value,
      onChange,
    } satisfies ChipSingleInputContextValue

    return (
      <ChipSingleInputContext.Provider value={contextValue}>
        <div ref={ref} className={cn('w-full', className)} {...props}>
          {label && (
            <div className='block text-sm font-medium text-foreground mb-3'>
              {label}
            </div>
          )}
          <div className='flex flex-wrap gap-2'>{children}</div>
          {hasError && (
            <div className='mt-2 text-sm text-danger' role='alert'>
              {error}
            </div>
          )}
          {!hasError && helperText && (
            <div className='mt-2 text-sm text-gray-600'>{helperText}</div>
          )}
        </div>
      </ChipSingleInputContext.Provider>
    )
  }
)

export const ChipSingleInput = Object.assign(ChipSingleInputRoot, {
  Option: ChipSingleInputOption,
})
