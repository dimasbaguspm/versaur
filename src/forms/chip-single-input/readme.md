
# ChipSingleInput

Single-select chip group for forms, using radio input pattern. Follows Versaur UI conventions for accessibility, theming, and composition.

## Features
- Single selection (radio pattern)
- Customizable color variants, shape, and size
- Optional custom check icon per option
- Accessible, keyboard navigable, mobile-first
- Compound pattern with context for shared state

## Usage
```tsx
import { ChipSingleInput } from 'versaur/forms/chip-single-input'

function Example() {
  const [value, setValue] = useState<string>('apple')
  return (
    <ChipSingleInput
      name="fruit"
      value={value}
      onChange={setValue}
      variant="primary"
      label="Select a fruit"
      helperText="Choose one"
    >
      <ChipSingleInput.Option value="apple">Apple</ChipSingleInput.Option>
      <ChipSingleInput.Option value="banana">Banana</ChipSingleInput.Option>
      <ChipSingleInput.Option value="cherry">Cherry</ChipSingleInput.Option>
    </ChipSingleInput>
  )
}
```

## Props
### `<ChipSingleInput />`
- `name` (string, required): Name for the radio group
- `value` (string, required): Selected value
- `onChange` (function, required): Callback for value changes
- `variant` (string): Color variant (primary, secondary, etc.)
- `shape` ('circle' | 'rounded'): Chip shape (default: 'circle')
- `size` ('sm' | 'md' | 'lg'): Chip size (default: 'sm')
- `label` (ReactNode): Optional label above the group
- `helperText` (ReactNode): Optional helper below the group
- `error` (ReactNode): Error message for invalid state

### `<ChipSingleInput.Option />`
- `value` (string, required): Option value
- `children` (ReactNode, required): Option label
- `check` (ReactNode): Custom check icon
- `defaultCheck` (boolean): Show default check icon

## Accessibility
- Uses radio input for native accessibility
- Keyboard navigation and ARIA attributes included

## Theming
- All variants, shapes, and sizes use Versaur design system and Tailwind
