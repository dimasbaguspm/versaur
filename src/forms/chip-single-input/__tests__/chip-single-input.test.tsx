import { render, screen, fireEvent } from '@testing-library/react'
import { composeStories } from '@storybook/react'
import * as stories from '../chip-single-input.stories'

describe('ChipSingleInput', () => {
  const {
    Basic,
    Sizes,
    States,
    IconOnly,
    WithIcons,
    Truncation,
    WithValidation,
  } = composeStories(stories)

  it('renders options, toggles selection, and displays correct text', () => {
    render(<Basic />)
    const apple = screen.getByLabelText('Apple')
    const banana = screen.getByLabelText('Banana')
    const cherry = screen.getByLabelText('Cherry')

    expect(apple).toBeInTheDocument()
    expect(banana).toBeInTheDocument()
    expect(cherry).toBeInTheDocument()

    // Banana should be checked initially
    expect(banana).toBeChecked()

    fireEvent.click(apple)
    expect(apple).toBeChecked()
    expect(banana).not.toBeChecked()

    fireEvent.click(cherry)
    expect(cherry).toBeChecked()
    expect(apple).not.toBeChecked()
  })

  it('renders size variations', () => {
    render(<Sizes />)
    const optionA = screen.getByLabelText('Option A')
    const optionC = screen.getByLabelText('Option C')
    const optionE = screen.getByLabelText('Option E')

    expect(optionA).toBeInTheDocument()
    expect(optionC).toBeInTheDocument()
    expect(optionE).toBeInTheDocument()

    fireEvent.click(optionA)
    expect(optionA).toBeChecked()
  })

  it('respects disabled and readOnly states', () => {
    render(<States />)

    // Test disabled state
    const optionB = screen.getByLabelText('Option B')
    expect(optionB).toBeDisabled()
    expect(optionB).toBeChecked() // Initially checked
    const disabledLabel = screen.getByText('Option B').closest('label')
    expect(disabledLabel?.className).toMatch(/cursor-not-allowed/)

    // Test readOnly state
    const optionD = screen.getByLabelText('Option D')
    expect(optionD).toHaveAttribute('readOnly')
    expect(optionD).toBeChecked() // Initially checked
    const readOnlyLabel = screen.getByText('Option D').closest('label')
    expect(readOnlyLabel?.className).toMatch(/cursor-default/)
  })

  it('renders icon-only chips', () => {
    render(<IconOnly />)
    const star = screen.getByLabelText('Star')
    const heart = screen.getByLabelText('Heart')
    const sparkles = screen.getByLabelText('Sparkles')

    expect(star).toBeInTheDocument()
    expect(heart).toBeInTheDocument()
    expect(sparkles).toBeInTheDocument()
    expect(heart).toBeChecked() // Initially checked

    fireEvent.click(star)
    expect(star).toBeChecked()
    expect(heart).not.toBeChecked()
  })

  it('renders chips with icons and text', () => {
    render(<WithIcons />)
    const star = screen.getByText('Star')
    const heart = screen.getByText('Heart')
    const sparkles = screen.getByText('Sparkles')

    expect(star).toBeInTheDocument()
    expect(heart).toBeInTheDocument()
    expect(sparkles).toBeInTheDocument()
  })

  it('renders truncation with maxWidth', () => {
    render(<Truncation />)
    const shortChip = screen.getByText('Short')
    const longChip = screen.getByText('Very Long Text That Will Be Truncated')

    expect(shortChip).toBeInTheDocument()
    expect(longChip).toBeInTheDocument()

    const longLabel = longChip.closest('label')
    expect(longLabel?.style.maxWidth).toBe('120px')
  })

  it('displays required asterisk, error and helper text', () => {
    render(<WithValidation />)

    // Check for required asterisk
    const asterisk = screen.getByText('*')
    expect(asterisk).toBeInTheDocument()
    expect(asterisk.className).toMatch(/text-danger/)

    // Check for error message
    const errorMessage = screen.getByText('Please select an option')
    expect(errorMessage).toBeInTheDocument()
    expect(errorMessage).toHaveAttribute('role', 'alert')

    // Check for helper text
    const helperText = screen.getByText(
      'Choose the option that best fits your needs'
    )
    expect(helperText).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<Basic />)
    expect(asFragment()).toMatchSnapshot()
  })
})
