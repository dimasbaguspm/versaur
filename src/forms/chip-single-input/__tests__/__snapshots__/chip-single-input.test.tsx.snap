// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChipSingleInput > matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Select fruits"
    class="w-full"
  >
    <div
      class="flex flex-wrap gap-2"
    >
      <input
        class="sr-only"
        id="«rr»"
        name="fruits"
        type="radio"
        value="apple"
      />
      <label
        class="inline-flex items-center bg-white transition-colors duration-200 cursor-pointer border border-border hover:bg-primary-soft hover:border-primary-bold focus:border-primary-light data-[selected=true]:bg-primary data-[selected=true]:border-primary data-[selected=true]:text-white rounded-full h-7 px-3 text-sm min-w-[2.25rem]"
        data-selected="false"
        for="«rr»"
      >
        <span
          class="flex items-center gap-2"
        >
          Apple
        </span>
      </label>
      <input
        class="sr-only"
        id="«rs»"
        name="fruits"
        type="radio"
        value="banana"
      />
      <label
        class="inline-flex items-center bg-white transition-colors duration-200 cursor-pointer border border-border hover:bg-primary-soft hover:border-primary-bold focus:border-primary-light data-[selected=true]:bg-primary data-[selected=true]:border-primary data-[selected=true]:text-white rounded-full h-7 px-3 text-sm min-w-[2.25rem]"
        data-selected="false"
        for="«rs»"
      >
        <span
          class="flex items-center gap-2"
        >
          Banana
        </span>
      </label>
      <input
        checked=""
        class="sr-only"
        id="«rt»"
        name="fruits"
        type="radio"
        value="cherry"
      />
      <label
        class="inline-flex items-center bg-white transition-colors duration-200 cursor-pointer border border-border hover:bg-primary-soft hover:border-primary-bold focus:border-primary-light data-[selected=true]:bg-primary data-[selected=true]:border-primary data-[selected=true]:text-white rounded-full h-7 px-3 text-sm min-w-[2.25rem]"
        data-selected="true"
        for="«rt»"
      >
        <span
          class="flex items-center gap-2"
        >
          Cherry
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
